### PR TITLE
update recipe for sequana 0.8.2

### PR DESCRIPTION
**Pull request by cokelaer**
**Major update of sequana recipes from 0.7 to 0.8.2**
----

> Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences**
      (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

> Everyone has access to the following BiocondaBot commands, which can be given in a comment:
>
>  * `@BiocondaBot please update` will cause the BiocondaBot to merge the master branch into a PR
>  * `@BiocondaBot please add label` will add the `please review & merge` label.
>  * `@BiocondaBot please fetch artifacts` will post links to packages and docker containers built by the CI system. You can use this to test packages locally before merging.
>
> For members of the Bioconda project, the following command is also available:
>
>  * `@BiocondaBot please merge` will cause packages/containers to be uploaded and a PR merged. Someone must approve a PR first! This has the benefit of not wasting CI build time required by manually merging PRs.
>
> If you have questions, please post them in gitter or ping `@bioconda/core` in a comment (if you are not able to directly ping `@bioconda/core` then the bot will repost your comment and enable pinging).

### CODE REVIEW

File: recipes/sequana/meta.yaml [hunk 1]
```diff
@@ -1,76 +1,88 @@
 {% set name = "sequana" %}
-{% set version = "0.7.1" %}
-{% set sha = "891e9a27fd992a9713de2e3afac9b4c8042d492703be65e1ae2b0ab098e275d6" %}
+{% set version = "0.8.2" %}
+{% set file_ext = "tar.gz" %}
```
Comment (marcelm) [inline recipes/sequana/meta.yaml]: Please don’t introduce the `file_ext` and `hash_type` Jinja2 variables as they just complicate the recipe. Jinja2 variables should only be added if they are used at least twice.

**Comment by cokelaer** [comment]
@marcelm  I'm going to close this pull request. I do not manage to build this recipes for now. Too many issues right. One issue is with bioservices, a software of mine, that I will update first. Another issue is with cython that is in my requirements but not really required. So I need to cleanup sequana first. The final issue is with multiqc, probably a new version that breaks this recipe.

**Comment by marcelm** [comment]
I added `cython` and  `xmldict` to the run requirements and the test within `conda build` actually pass now. However, it seems that building the container in `mulled-build` seems to be stuck at `Collecting package metadata`. Either there’s some intermediate problem or the resolver is overwhelmed because the recipe has so many dependencies. You may want to try adding some version constraints to some of the dependencies so the resolver doesn’t have to work as much. This will also benefit every user who installs the package :-).

**Comment by marcelm** [comment]
Tests are not running; I’ll try to trigger them.

**Comment by cokelaer** [comment]
@marcelm thanks for the different fixes. locally, it seems to resolve the deps in a reasonnable amount of time. The restrictions you added seem good to me. 

Another point but I do not think this is related: I know that pyqt may cause trouble currently with some version of python (https://github.com/sequana/sequana/issues/574, https://github.com/mantidproject/mantidimaging/issues/339) in particular, I believe that I should build this recipe with a restriction on python<=3.7.3 so I will need to change that in the recipe.

**Comment by marcelm** [comment]
> I know that pyqt may cause trouble currently with some version of python

The `matplotlib-base` dependency doesn’t pull in `pyqt` (see #19996), so perhaps you don’t have that problem. If you actually need PyQt, I recommend that you do not change the dependency back to regular `matplotlib` (because that will trigger the linter), but to add `pyqt` as an explicit dependency.

**Comment by cokelaer** [comment]
@marcelm the linux build is successful but not the osx. Do you have any idea about the error message ? I do not see any obvious reason for the failed build. What would you suggest ? thanks in advance

**Comment by marcelm** [comment]
I think the error was: `...placehold_placehold_p/bin/python: No module named pip` (but the log output is hard to read)

**Comment by cokelaer** [comment]
I see it now, okay tough one. Let us see now. thanks again

**Comment by marcelm** [comment]
@BiocondaBot please fetch artifacts

**Comment by cokelaer** [comment]
@marcelm thanks so much for your help. This recipe was not easy indeed. I have now merged it.

**Comment by marcelm** [comment]
Yes, this was quite a bit of work I can imagine, glad it worked out in the end :-).

**Comment by dpryan79** [comment]
Nice work everyone!

**Pull request merged**